bin/main.ml

```diff
@@ -816,21 +816,49 @@ let exec =
   in
   let man =
     [ `S "DESCRIPTION"
-    ; `P {|$(b,jbuilder exec -- COMMAND) should behave in the same way as if you do:|}
+    ; `P {|$(b,jbuilder exec -- COMMAND) when COMMAND doesn't have a / will
+           behave in the same way as if you do:|}
     ; `Pre "  \\$ jbuilder install\n\
            \  \\$ COMMAND"
     ; `P {|In particular if you run $(b,jbuilder exec ocaml), you will have access
            to the libraries defined in the workspace using your usual directives
            ($(b,#require) for instance)|}
+    ; `P {|When a leading / is present in the command (absolute path), then the
+           path is interpreted relative to the build path of the specified
+           context|}
+    ; `P {|When a / is present at any other position (relative path), then the
+           path is interpeted as relative to the specified build context +
+           current working directory|}
     ; `Blocks help_secs
     ]
   in
   let go common context prog args =
+    let runcwd = Sys.getcwd () in
     set_common common ~targets:[];
     let log = Log.create () in
     let setup = Future.Scheduler.go ~log (Main.setup ~log common) in
     let context = Main.find_context_exn setup ~name:context in
-    let path = Config.local_install_bin_dir ~context:context.name :: context.path in
+    let (prog, path) =
+      match String.index prog '/' with
+      | None ->
+        (prog, Config.local_install_bin_dir ~context:context.name :: context.path)
+      | Some i ->
+        let p = Path.of_string prog in
+        let path =
+          if i = 0 then (
+            String.drop_prefix prog ~prefix:"/"
+            |> Option.value_exn
+            |> Path.of_string
+            |> Path.parent
+            |> Path.append context.build_dir
```

Comment:
There's reason in this, but IMO this subtlety will be lost on the majority of users and might even cause some confusion.

By the way, another variation to consider: only amend the _build/default/install/bin to the PATH in every case but still maintain behavior 2 and 3 with an absolute path to the binary.

Reply:
> Sorry, I don't get what you mean by that. Do you agree to use directly the path in case 3 or do you prefer your behavior?

I just meant that I don't like how the PATH ends up varying in the 3 different cases (2 cases in my current implementation). Even though we have some good reasons for it.

Seems like we agree that it's best to keep the PATH constant and simply locate the binary ourselves. So I'll do that.

Reply:
I just misunderstood your first comment. Just pushed a commit that implements things as you've said.